### PR TITLE
Suppress redundant linkage warnings due to dependency linkage propagation

### PR DESCRIPTION
**Pull request by pusukuri**
Suppress redundant linkage warnings due to dependency linkage propagation rdar://112671586

### Motivation:

Improve user experience by suppressing  redundant linkage warnings

### Modifications:

Pass "-no_warn_duplicate_libraries" to linker.

### Result:

We should not see warnings such as "ld: warning: ignoring duplicate libraries: '-lsqlite3'"

### CODE REVIEW

**Comment by pusukuri** [comment]
@swift-ci test

**Comment by pusukuri** [comment]
@swift-ci test macos

File: Tests/BuildTests/BuildSystemDelegateTests.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ final class BuildSystemDelegateTests: XCTestCase {
             try XCTSkipIf(true, "test is only supported on macOS")
             #endif
             let (fullLog, _) = try executeSwiftBuild(fixturePath)
-            XCTAssertTrue(fullLog.contains("ld: warning: ignoring duplicate libraries: '-lz'"), "log didn't contain expected linker diagnostics")
+            XCTAssertFalse(fullLog.contains("ld: warning: ignoring duplicate libraries: '-lz'"), "log contains unnecessary linker warnings")
```
Comment (neonichu) [inline Tests/BuildTests/BuildSystemDelegateTests.swift:28]: I don't think it's correct to change this test, it was supposed to ensure that linker warnings are properly propagated. We'll need to find a different linker warning that we can check for here or remove the test if appropriate.

Comment (pusukuri) [reply Tests/BuildTests/BuildSystemDelegateTests.swift:28]: Thank you @neonichu  I am thinking about removing the test case.

Comment (MaxDesiatov) [reply Tests/BuildTests/BuildSystemDelegateTests.swift:28]: I agree, with the current change we're losing test coverage here. This test needs to change to check for a different warning.

**Comment by pusukuri** [comment]
@swift-ci test

**Comment by pusukuri** [comment]
@swift-ci test macos

File: Tests/BuildTests/BuildSystemDelegateTests.swift (hunk 1 above)
Comment (pusukuri) [reply Tests/BuildTests/BuildSystemDelegateTests.swift]: Talked with @MaxDesiatov and made changes appropriately for the above test case.

File: Fixtures/Miscellaneous/DoNotFilterLinkerDiagnostics/Package.swift [hunk 2]
```diff
@@ -8,8 +8,7 @@ let package = Package(
         .executableTarget(
             name: "DoNotFilterLinkerDiagnostics",
             linkerSettings: [
-                .linkedLibrary("z"),
-                .unsafeFlags(["-lz"]),
+                .unsafeFlags(["-Lfoobar"]),
                 // should produce: ld: warning: ignoring duplicate libraries: '-lz'
```
Comment (MaxDesiatov) [inline Fixtures/Miscellaneous/DoNotFilterLinkerDiagnostics/Package.swift:12]: This comment is no longer relevant.
```suggestion
```

**Review by MaxDesiatov — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by pusukuri** [comment]
@swift-ci test windows

**Pull request merged**